File: 8KYU/Find nearest square number/solution.js
function nearestSq(n){
  var sqrt = Math.sqrt(n);
  var floor = Math.floor(sqrt);
  var ceiling = Math.ceil(sqrt);  
  var floorDiff = Math.abs(n - (floor*floor));
  var ceilingDiff = Math.abs(n - (ceiling*ceiling));
  if(floorDiff > ceilingDiff)
    return ceiling*ceiling;
  return floor*floor;
}

File: 8KYU/Find nearest square number/solution.c
int nearest_sq(int n)
{
  double sq = sqrt(n);
  double f = floor(sq);
  double c = ceil(sq);
  
  int fDiff = abs(n - f * f);
  int cDiff = abs(n - c * c);
  
  if(fDiff > cDiff)
    return c * c;
  
  return f * f;
}